catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -742,4 +750,35 @@ private boolean checkSingleFile(Fileset fileset) {
           fileset.name());
     }
   }
+
+  FileSystem getFileSystem(Path path, Map<String, String> config) throws IOException {
+    if (path == null) {
+      throw new IllegalArgumentException("Path should not be null");
+    }
+
+    // Can't get the scheme from the path like '/path/to/file', use the default filesystem provider.
+    if (path.toUri().getScheme() == null) {
+      if (defaultFilesystemProvider != null) {
+        return getFileSystemByScheme(defaultFilesystemProvider, config, path);
+      }
+
+      LOG.warn(
+          "Can't get schema from path: {} and default filesystem provider is null, using"
+              + " local file system",
+          path);
+      return getFileSystemByScheme(LOCAL_FILE_SCHEME, config, path);
+    }
+
+    return getFileSystemByScheme(path.toUri().getScheme(), config, path);
+  }
+
+  private FileSystem getFileSystemByScheme(String scheme, Map<String, String> config, Path path)
+      throws IOException {
+    FileSystemProvider provider = fileSystemProvidersMap.get(scheme);
+    if (provider == null) {
+      throw new IllegalArgumentException("Unsupported scheme: " + scheme);
```

Comment:
You should clearly tell user why the exception is happened, the code here is not easy for user to know the actual reason.